README.md

```diff
@@ -1,138 +1,615 @@
+
+```mjs
+import { @frozen } from "./frozen.mjs";
+
+@frozen
+class MyClass {
+  method() { }
+}
+
+MyClass.method = () => {};            // TypeError to add a method
+MyClass.prototype.method = () => {};  // TypeError to overwrite a method
+MyClass.prototype.method.foo = 1;     // TypeError to mutate a method
+```
+
+`@frozen` is implemented with the `@register` decorator, which allows a callback to be scheduled after the class is created. The callback is passed the class as an argument.
+
+```mjs
+// frozen.mjs
+export decorator @frozen {
+  @register(klass => {
+    Object.freeze(klass);
+    for (const [key, value] of Object.entries(klass)) {
+      Object.freeze(value);
+    }
+    for (const [key, value] of Object.entries(klass.prototype)) {
+      Object.freeze(value);
+    }
+  })
+}
+```
+
+### `@set`
+
+The `@set` decorator makes a class field declaration behave as a setting a property when it's called, rather than as `Object.defineProperty`. In particular, setters will be called with a normal property set, whereas `Object.defineProperty` just clobbers setters without calling them. For example:
+
+```mjs
+class SuperClass {
+  set x(value) { console.log(value); }
+}
+
+class SubClassA extends SuperClass {
+  x = 1;
+}
+
+class SubClassB extends SuperClass {
+  @set x = 1;
+}
+
+const a = new SubClassA();  // does not log anything
+a.x;                        // 1
+
+const b = new SubClassB();  // logs 1
+b.x;                        // undefined
+```
+
+The `@set` decorator is implemented with `@initialize`, which can decorate public fields. `@initialize` takes a callback as an argument, which is called after the field initializer is evaluated,
+
+```mjs
+// set.mjs
+
+export decorator @set { @initialize(function(value, key) { this[key] = value }) }
+```
+
+### `@tracked`
+
+The `@tracked` decorator turns a public field declaration into a getter/setter pair which triggers a `render()` method when the setter is called. This pattern, or patterns like it, is common in frameworksto avoid extra bookkeeping scattered throughout the application to ask for re-rendering.
+
+```mjs
+import { @tracked } from "./tracked.mjs";
+
+class Element {
+  @tracked counter = 0;
+
+  increment() { this.counter++; }
+
+  render() { console.log(counter); }
+}
+
+const e = new Element();
+e.increment();  // logs 1
+e.increment();  // logs 2
+```
+
+`@tracked` is defined in terms of the combination of two built-in decorators that we've seen before above. `@initialize` is used to replace the property definition with setting a property which stores the underlying data. `@register` is used to define a getter/setter pair which is used when accessing the property.
+
+```mjs
+// tracked.mjs
+
+export decorator @tracked {
+  @initialize(function(value, name) { this[`__internal_${name}`] = value; })
+  @register((target, name) => {
+    Object.defineProperty(target, "name", {
+      get() { return this[`__internal_${name}`]; },
```

Comment:
this seems like a perfect use case for decorator-controlled private fields.

Reply:
Yes, coming in NEXTBUILTINS.md.